go/private/binary.bzl

```diff
@@ -175,18 +168,24 @@ def emit_go_link_action(ctx, transitive_go_library_paths, transitive_go_librarie
   extldflags = c_linker_options(ctx) + [
       "-Wl,-rpath,$ORIGIN/" + ("../" * pkg_depth),
   ]
-  for d in cgo_deps:
-    if d.basename.endswith('.so'):
-      short_dir = d.dirname[len(d.root.path):]
-      extldflags += ["-Wl,-rpath,$ORIGIN/" + ("../" * pkg_depth) + short_dir]
 
   gc_linkopts, extldflags = _extract_extldflags(gc_linkopts, extldflags)
 
   link_opts = [
       "-L", "."
   ]
-  for path in transitive_go_library_paths:
-    link_opts += ["-L", path]
+  libs = depset()
+  cgo_deps = depset()
+  for golib in depset([library]) + library.transitive:
+    libs += [getattr(golib, "library" + mode)]
+    link_opts += ["-L", getattr(golib, "searchpath" + mode)]
```

Comment:
I started doing this, but I need to change a few places and it's hard to do cleanly before the cover cl lands as that one rearranges the source tranformation step to not need the output object pre declared, I would like to fix this in a follow on cl.